### ISSUE TITLE
lammpsdatareader: detect wrong atom type

### ISSUE THREAD

**Issue by junghans** [votca/csg#537]
Reported in https://groups.google.com/d/topic/votca/ubngBiT0lEs/discussion

In short, `csg_dump --top spce.data` with input from [here](https://github.com/votca/csg-tutorials/blob/stable/spce/ibi_lammps/spce.data) normally leads to:
```
molecule: 1 Be1 beads: 1
0 Name Bead1 Type Bead1 Mass 18.0154 Resnr 1 Resname DUM Charge 0
molecule: 2 Be1 beads: 1
1 Name Bead1 Type Bead1 Mass 18.0154 Resnr 2 Resname DUM Charge 0
```
with the bead type being `Bead1` meaning that atom with the mass `18.0154` aren't match to an atom type, but on some machines the output will be:
```
List of molecules:
molecule: 1 F1 beads: 1
0 Name F1 Type F1 Mass 18.0154 Resnr 1 Resname DUM Charge 0
molecule: 2 F1 beads: 1
1 Name F1 Type F1 Mass 18.0154 Resnr 2 Resname DUM Charge 0
```
which means the atom with mass `18.0154` got matched up with `F` at type (mass `18.9984032`), which is obviously wrong.

**Comment by JensWehner** [comment]
hmm weird... I guess the tolerance is set wrongly.

**Comment by JoshuaSBrown** [comment]
Ya the tolerance should not be that poor. I can't look tonight but probably can on Fri or Sat.

**Comment by JoshuaSBrown** [comment]
I have been unable to reproduce this error with v1.6 or v1.6_rc2, on wsl ubuntu 18. I can try tomorrow on ubuntu 16.

**Comment by junghans** [comment]
You can ask on the mailing list for details as well. I haven't see this issue before either, I was guessing 32-bit, but that wasn't the case.

Anyhow, I think automatically assigning atom types doesn't make a lot of sense for coarse-grained topologies.

**Comment by JoshuaSBrown** [comment]
> Anyhow, I think automatically assigning atom types doesn't make a lot of sense for coarse-grained topologies.

I think it is useful for lammps, we could make it an optional flag though, --auto-detect.

**Comment by JoshuaSBrown** [comment]
I found the problem:

>Returns the absolute value of x: |x|.
>
>These convenience abs overloads are exclusive of C++. In C, abs is only declared in <stdlib.h> (and operates on int values).

Some where a c header file is being included. I think it would be a good policy to require the use of fabs instead of abs any time it is being used on a double to avoid this problem. Will submit a fix later.

**Comment by JensWehner** [comment]
no just use std::abs. This is the problem with using std;.

**Comment by junghans** [comment]
I think @JensWehner is right: https://stackoverflow.com/a/33738702

**Comment by junghans** [comment]
Fix in #705
